File: 12. Integer to Roman/12. Integer to Roman.rb
# @param {Integer} num
# @return {String}
def int_to_roman(num)
    values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
    symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
    result = ""
    
    (0...values.length).each do |i|
        while num >= values[i]
            result += symbols[i]
            num -= values[i]
        end
    end
    
    result
end

File: 12. Integer to Roman/12. Integer to Roman.dart
class Solution {
  String intToRoman(int num) {
    List<int> values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
    List<String> symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"];
    StringBuffer result = StringBuffer();
    
    for (int i = 0; i < values.length && num > 0; i++) {
      while (num >= values[i]) {
        result.write(symbols[i]);
        num -= values[i];
      }
    }
    
    return result.toString();
  }
}